lib/PrincipleStudios.OpenApiCodegen.Client.TypeScript.Test/Utils/CommonDirectoryFixture.cs

```diff
@@ -111,6 +111,19 @@ private async Task GenerateTypeScriptFrom(string documentName)
 
             File.WriteAllText(Path.Combine(generatedFolder, entry.Key), contents: entry.SourceText);
         }
+
+        // Ensure the codegenerated files build
+        var tscResult = await NodeUtility.Tsc(SetupProcess, CancellationToken);
+        if (tscResult.ExitCode != 0)
+        {
+            var tsconfig = Path.Combine(DirectoryPath, "tsconfig.json");
+            var tsconfigContents = System.Text.Json.JsonSerializer.Deserialize<System.Text.Json.Nodes.JsonObject>(File.ReadAllText(tsconfig));
+            var exclude = (System.Text.Json.Nodes.JsonArray)(tsconfigContents!["exclude"] ??= new System.Text.Json.Nodes.JsonArray());
+            exclude.Add(documentName + "/");
+            File.WriteAllText(tsconfig, tsconfigContents.ToJsonString());
```

Comment:
This block ensures that failed folders get excluded from future tsc compilation without deleting the files - this way the dev can go look at the failed files and check the output!